UN-OCHA/unocha-site
# [UNO-593] Videos

Refs: UNO-593

Note: I got the ticket number wrong so the commits don't reference the proper one...

This PR adds the option to select a video from Canto (in addition to youtube) when using a video paragraph type using the same widget as the image media type.

<img width="1015" alt="Screenshot 2023-05-17 at 13 00 13" src="https://github.com/UN-OCHA/unocha-site/assets/696348/8929a909-e3fb-4d6a-8abc-7c3e8b3e6245">

It uses the https://www.drupal.org/project/oembed_providers module to add a custom provide for the UNOCHA Canto videos and adds the corresponding route and logic so that it Canto videos are handled the same way as Youtube videos.

## Styling

There is no styling in this PR. The video is rendered via a `<video>` (see [code](https://github.com/UN-OCHA/unocha-site/blob/7e2532e90d8f00ce740138e9d54f441e41d7b188/html/modules/custom/unocha_canto/src/Controller/CantoController.php#L140)) via an iframe (default Drupal oembed's behavior for security). 

The iframe has the `width` and `height` of the original video. The wrapping field element uses this rule https://github.com/UN-OCHA/unocha-site/blob/7e2532e90d8f00ce740138e9d54f441e41d7b188/html/themes/custom/common_design_subtheme/css/styles.css#L276 to restrict the dimensions and make it responsive.

The problem is that the ratio is fixed in that CSS rule. I think we could compute it based on the width and height of the video in a `field--media--field-media-oembed-video.html.twig` template for example. @left23 What do you think?

Also, the above rule should be added to the claro admin subtheme so that it's used for the video paragraph previews.

### Tests

1. Recreate the container image or run `composer install` to add the oembed_providers menu
2. Clear the cache and import the config
3. Edit/create a page and add a video paragraph type
4. Click select from Canto to see the list of available video and select one
5. Check that the video appears in the paragraph preview in the form
6. Save
7. Check that the video is rendered correctly
